src/services/QueryService.ts

```diff
@@ -36,15 +43,18 @@ export class QueryService<T> {
   }
 
   get total(): Promise<number> {
-    if (this._lastResponse && this._lastResponse.total) {
-      return Promise.resolve(this._lastResponse.total);
-    }
-    return this.http
-      .get(this.endpoint, { params: { fields: 'id', count: 0, ...this.parameters } })
-      .then((response: AxiosResponse) => response.data)
-      .then((result: BullhornListResponse<T>) => {
-        return result.total || 0;
-      });
+    // tslint:disable-next-line:promise-function-async
+    return this.initialized.then(() => {
+      if (this._lastResponse && this._lastResponse.total) {
```

Comment:
I used this because I couldn't figure out the notation to use async with a getter. I dug into it and was able to figure it out. Let me know what you think.

Reply:
good old IFFEs